### ISSUE TITLE
Detect Mining range from Tooltip and stop ship at maximum range-1km

### ISSUE THREAD

**Issue by habeas81** [Viir/bots#27]
I'd recommend that the ship stays at mining range so it doesn't get hung up on asteroids if it needs to escape.  Is it possible to expose the fields as a configuration item for user-input in the script, or otherwise have the range detected from the mining module tooltip and have the ship approach until it is inside the mining range, then stop and activate mining modules? 

Thanks!

**Comment by Viir** [comment]
Sounds interesting; let's have a closer look at your setup.

What reading do you get from the module button tooltip? Do you see that range in there?

About the possibility of stopping the approach: Maybe you have a button in the ship UI to achieve this effect? Can you look at the available actions there to see if something helps you with that effect?

**Comment by habeas81** [comment]
Yes the mining range is listed in the tooltip [see image](https://pasteboard.co/JDfL2qJ.png) and the ship status cluster [see image](https://pasteboard.co/JDfLwha.png) has a button for "stop the ship".  It is the "-" symbol on the left side of the ship speed indicator window.

**Comment by Viir** [comment]
> Yes the mining range is listed in the tooltip [see image](https://pasteboard.co/JDfL2qJ.png)

In this screenshot, I see a text `Range within 19 km` left of the range. How about searching for the text `Range within` in the reading from the game client to find the range?

**Comment by habeas81** [comment]
Will do, though I'm very new at this so it might take me awhile to figure out how to read and interpret the memory using the tools and guides you have so kindly provided!  Thanks!

**Comment by Viir** [comment]
You don't need to read the memory or interpret it. You can upload the file with your session as-is. When I have your session file, I check the question about `getAllContainedDisplayTexts` on my end. The same applies to the stop button, I think.

As far as I know, you can upload a file directly here in the GitHub issue. Others have already used this successfully, like here: https://github.com/Viir/bots/issues/23

The process to export and upload usually takes less than five minutes, but of course, it depends on your network bandwidth.

**Comment by Mitsuakisan** [comment]
Has this been addressed already?  I was playing around with the newest release and noticed that after it finished mining out an asteroid it would immediately lock another and approach, however it would not orbit or stop the ship when within range. This means you fly past it and move infinitely away while the bot, goes into a loop of trying to turn on the mining equips even if they are out of range. There needs to be an orbit command where after locking instead of approaching it orbits instead. That would eliminate all of these issues.

**Comment by Viir** [comment]
@Mitsuakisan that sounds interesting. I will take a look at your scenario, too, if you can share it.
